ethclient/ethclient_test.go

```diff
@@ -168,15 +168,24 @@ var (
 	testBalance = big.NewInt(2e10)
 )
 
-func newTestBackend(t *testing.T) (*node.Node, []*types.Block) {
-	// Generate test chain.
-	genesis, blocks := generateTestChain()
+func newTestBackend(t *testing.T, withTx bool) (*node.Node, []*types.Block) {
+	var (
+		genesis *core.Genesis
+		blocks  []*types.Block
+	)
+
+	// Generate test chain
+	if withTx {
+		genesis, blocks = generateTestChainWithTx(t)
+	} else {
+		genesis, blocks = generateTestChain()
+	}
 
 	// Start Ethereum service.
 	var ethservice *eth.Ethereum
 	n, err := node.New(&node.Config{})
 	n.Register(func(ctx *node.ServiceContext) (node.Service, error) {
-		config := &eth.Config{Genesis: genesis}
+		config := &eth.Config{Genesis: genesis, NoPrefetch: true}
```

Comment:
without this parameter it produces `nonce too high` error. For details see #19723